modules/post/multi/manage/local_mitm.rb

```diff
@@ -0,0 +1,58 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+  include Msf::Post::Windows::Priv
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'          => 'Local MITM',
+      'Description'   => %q{
+        This module setup a local proxy on the victim computer so you can MITM it.
+      },
+      'License'       => MSF_LICENSE,
+      'Author'        => [ 'Eliott Teissonniere' ],
+      'Platform'      => [ 'win', 'linux' ],
+      'SessionTypes'  => [ 'meterpreter', 'shell' ]
+    ))
+
+    register_options(
+      [
+        OptString.new('RHOST', [true, 'Host of the socks proxy server']),
+        OptString.new('RPORT', [true, 'Port of the socks proxy server']),
+        OptBool.new('CLEANUP', [false, 'If we should remove the proxy instead of installing it'])
+      ])
+  end
+
+  def run
+    if session.platform == 'windows'
+      print_status('Windows detected, using netsh')
+
+      unless datastore['CLEANUP']
```

Comment:
See comment above about using module ACTIONs.

However, should you decide not to go that route...

The preferred structure for conditional statements is to have the positive logic as the fist branch, rather than starting with `unless`.

Also, it might be nice to split the mitm for each platform into a different method. This would be cleaner.

Reply:
Will use actions if you think it is better 🙂

And yes, this is planned (splitting the code), will refactor that huge monolithic block as soon as I can guarantee it all works as expected.